blockmanager.go

```diff
@@ -223,6 +223,24 @@ func (b *blockManager) startSync(peers *list.List) {
 		enext = e.Next()
 		sp := e.Value.(*serverPeer)
 
+		// Once the segwit soft-fork package has activated, we only
+		// want to sync from peers which are witness enabled to ensure
+		// that we fully validate all blockchain data.
+		segwitState, err := b.chain.ThresholdState(chaincfg.DeploymentSegwit)
+		if err != nil {
+			bmgrLog.Errorf("Unable to query for segwit "+
+				"soft-fork state: %v", err)
+			continue
+		}
+		segwitActive := segwitState == blockchain.ThresholdActive
+		sp.witnessMtx.Lock()
+		if segwitActive && !sp.witnessEnabled {
+			bmgrLog.Infof("peer %v not witness enabled, skipping", sp)
```

Comment:
Nice observation. I've added the help function which ensures that the proper lock is always obtained before accessing the variable.